llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp

```diff
@@ -7143,6 +7143,22 @@ void SelectionDAGBuilder::visitIntrinsicCall(const CallInst &I,
     setValue(&I, DAG.getNode(ISD::ABS, sdl, Op1.getValueType(), Op1));
     return;
   }
+  case Intrinsic::scmp: {
+    SDValue Op1 = getValue(I.getArgOperand(0));
+    SDValue Op2 = getValue(I.getArgOperand(1));
+    EVT DestVT = DAG.getTargetLoweringInfo().getValueType(DAG.getDataLayout(),
```

Comment:
```suggestion
    EVT DestVT = TLI.getValueType(DAG.getDataLayout(),
```

See https://github.com/llvm/llvm-project/blob/4b5e0a1d12f0d624852c6d7c82207647ed5282aa/llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp#L6342-L6346